higher/optim.py

```diff
@@ -212,6 +214,8 @@ def step(
             allow_unused=True  # boo
         )
 
+        self._grads_preproc = [grad.clone() for grad in all_grads]
```

Comment:
What is this line doing? What is the purpose of `_grad_preproc`, and do we want to be cloning everything every time the optimizer is called?

Reply:
I'm not so sure about the rest of the changelist: when I implemented this, I just had the callback applied to `all_grads` here, which has the additional benefit that the callback and subselect which grads to modify, rather than modify them all. Why not do something like that here? For the record, given the added argument
`grad_callback: _typing.Optional[_GradCallbackType] = None` to the `step` function definition, where `_GradCallbackType = _typing.Callable[[_typing.List[_torch.Tensor]], _typing.List[_torch.Tensor]]`, I would just write
```python
if grad_callback is not None:
            all_grads = grad_callback(all_grads)
        elif self._grad_callback is not None:
            all_grads = self._grad_callback(all_grads)
```
instead of this line cloning stuff and be good to go...